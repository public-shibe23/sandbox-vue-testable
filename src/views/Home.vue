<template>
  <div>
    <ProductList :products="products" @fetch="fetchProducts"/>
    <button @click="onclick">ckick</button>
  </div>
</template>

<script>
// @ is an alias to /src
import ProductList from "@/components/ProductList.vue";
import { mapState } from "vuex";

export default {
  name: "home",
  components: {
    ProductList
  },
  computed: mapState({
    products: state => state.ProductList.products,
  }),
  methods: {
    fetchProducts() {
      this.$store.dispatch("ProductList/FETCH_PRODUCTS");
    },

    getTotal(products) {
      return products.reduce((accum, current) => {
        accum + current;
      });
    },

    onclick() {
      console.log(this.$store);
    }
  }
};
</script>
