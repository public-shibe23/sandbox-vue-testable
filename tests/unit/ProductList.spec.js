import { mount, createLocalVue } from "@vue/test-utils";
import Vuex from "vuex";
import Home from "@/views/Home.vue";
import ProductListModule from "@/store/ProductList";

const localVue = createLocalVue();

localVue.use(Vuex);

describe("ProductList.vue", () => {
  let actions;
  let state;
  let store;
  let wrapper;

  beforeEach(() => {
    state = {
      products: [
        {
          id: 1000,
          name: "T-shirts",
          price: 980,
          stock: 20
        }
      ]
    };

    actions = {
      FETCH_PRODUCTS: jest.fn()
    };

    store = new Vuex.Store({
      modules: {
        ProductList: {
          namespaced: true,
          state,
          actions,
          getters: ProductListModule.getters
        }
      }
    });
    wrapper = mount(Home, { store, localVue });
  });

  it("Clickボタンを押すと、actions.FETCH_PRODUCTSが呼ばれる", () => {
    const button = wrapper.find(".button");
    console.log(wrapper.text());
    button.trigger("click");
    expect(actions.FETCH_PRODUCTS).toHaveBeenCalled();
  });

  it("match snapshot", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
