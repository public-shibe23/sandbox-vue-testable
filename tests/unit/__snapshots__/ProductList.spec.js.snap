// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductList.vue match snapshot 1`] = `
<div>
  <div><button class="button">test!</button>
    <p>T-shirts</p>
  </div> <button>store</button>
</div>
`;
